--- mupdf_orig/source/tools/mudraw.c	Fri Oct 30 05:06:14 2015
+++ mupdf/source/tools/mudraw.c	Mon Jun 13 00:13:01 2016
@@ -826,6 +826,15 @@
 	return &p[1];
 }
 
+int Interactive(int argc, char **argv);
+int InteractiveCheck(int argc, char **argv) {
+	int i = 0;
+	for (i = 0; i < argc; ++i) {
+		if (lstrcmp(argv[i], "--interactive") == 0) return 1;
+	}
+	return 0;
+}
+
 int main(int argc, char **argv)
 {
 	char *password = "";
@@ -833,6 +842,10 @@
 	int c;
 	fz_context *ctx;
 	fz_alloc_context alloc_ctx = { NULL, trace_malloc, trace_realloc, trace_free };
+
+	if (InteractiveCheck(argc, argv)) {
+		return Interactive(argc, argv);
+	}
 
 	fz_var(doc);
 
