--- svg-device.c.orig	Fri Oct 02 23:26:00 2015
+++ svg-device.c	Thu Dec 17 20:07:07 2015
@@ -47,6 +47,8 @@
 	int num_fonts;
 	int max_fonts;
 	font *fonts;
+
+	int in_tile;
 };
 
 /* SVG is awkward about letting us define things within symbol definitions
@@ -134,7 +136,7 @@
 	if (ctm->a != 1.0 || ctm->b != 0 || ctm->c != 0 || ctm->d != 1.0 || ctm->e != 0 || ctm->f != 0)
 	{
 		fz_printf(out, " transform=\"matrix(%g,%g,%g,%g,%g,%g)\"",
-			ctm->a, ctm->b, ctm->c, ctm->d, ctm->e, ctm->f);
+			ctm->a, ctm->b, ctm->c, ctm->d, sdev->in_tile ? 0 : ctm->e, sdev->in_tile ? 2 * sdev->tiles->step.y : ctm->f);
 	}
 }
 
@@ -938,6 +940,7 @@
 	t->pattern = sdev->id++;
 	t->step.x = xstep;
 	t->step.y = ystep;
+	sdev->in_tile = 1;
 
 	/* view = area of our reference tile in pattern space.
 	 * area = area to tile into in pattern space.
@@ -1020,6 +1023,7 @@
 	out = end_def(sdev);
 
 	/* Finally, fill a rectangle with the pattern. */
+	sdev->in_tile = 0;
 	fz_printf(out, "<rect");
 	svg_dev_ctm(sdev, &t->ctm);
 	fz_printf(out, " fill=\"url(#pa%d)\" x=\"%g\" y=\"%g\" width=\"%g\" height=\"%g\"/>\n",
@@ -1062,6 +1066,7 @@
 		sdev->out = out;
 		sdev->out_store = out;
 		sdev->id = 0;
+		sdev->in_tile = 0;
 
 		dev = fz_new_device(ctx, sdev);
 	}
